pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -69,11 +69,16 @@ func (s *Service) InstanceByTags(machine *actuators.MachineScope) (*v1alpha1.Ins
 }
 
 // InstanceIfExists returns the existing instance or nothing if it doesn't exist.
-func (s *Service) InstanceIfExists(id string) (*v1alpha1.Instance, error) {
-	klog.V(2).Infof("Looking for instance %q", id)
+func (s *Service) InstanceIfExists(id *string) (*v1alpha1.Instance, error) {
+	if id == nil {
+		klog.Error("Instance does not have an instance id")
+		return nil, nil
```

Comment:
I think we want to return an error here, not nil.

Reply:
if we return an error then the function will continue to loop and wait for the instance id to exists....but if it never exists then the machine object won't get deleted

Reply:
Sorry to fork the conversation here. I responded https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/644#discussion_r264913317